applications/HDF5Application/custom_io/hdf5_file.h

```diff
@@ -85,216 +99,409 @@ class KRATOS_API(HDF5_APPLICATION) File
     File& operator=(File&& rOther);
 
     ///@}
-    ///@name Operations
+    ///@name Public operations
     ///@{
 
-    /// Check if path exists in HDF5 file.
+    /**
+     * @brief Checks if the given @ref rPath exists.
```

Comment:
Doxygen can't `@ref` locally defined tokens (such as function arguments or template parameters), only tokens that have their own entry in the docs (such as functions, classes, enums, etc.). Plz change these `@ref`s to something else (`@a` for example) in the rest of the docstrings too.

```suggestion
     * @brief Checks if the given @a rPath exists.
```

Reply:
done :)